### DISCUSSION TITLE
[MFA] Multi-Factor Authentification

### DISCUSSION THREAD

**Discussion by eprietotickamore** [apache/superset#29597] [Ideas]
I think it would be interesting to add a variable for Multi-Factor Authentication (MFA) in the configuration. 

Here is how MFA could work:

1. You enter your username and password.
2. The system checks if the credentials are correct. If they are, it sends an email with a one-time use code.
3. You then enter this code on another webpage that appears after the login page.

**Comment by sfirke** [comment]
Superset uses [Flask-AppBuilder's security module](https://flask-appbuilder.readthedocs.io/en/latest/security.html), so this would need to be added there. It's an interesting idea.

I am going to close this discussion; if this is still something you desire, or if anyone could contribute this (it's probably too big a lift to be a priority for the FAB maintainers), you could open an issue in their GitHub repo.

**Reply by ziggekatten** [reply]
This is normally done by using an idp like keycloak or ubisecure. We use ubisecure, and it is quite simple to set up.
